src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -253,9 +251,31 @@ internal RouteEndpointBuilder GetSingleRouteEndpointBuilder()
         }
 
         return builder;
+
+        [UnconditionalSuppressMessage("Trimmer", "IL2026",
+            Justification = "We surface a RequireUnreferencedCode in the call to the Map methods adding route handlers to this EndpointDataSource. Analysis is unable to infer this. " +
+            "Map methods that configure a RequestDelegate don't use trimmer unsafe features.")]
+        [UnconditionalSuppressMessage("AOT", "IL3050",
+            Justification = "We surface a RequiresDynamicCode in the call to the Map methods adding route handlers this EndpointDataSource. Analysis is unable to infer this. " +
+            "Map methods that configure a RequestDelegate don't use AOT unsafe features.")]
```

Comment:
Is there some way we can verify this?

One idea is to add a "Native AOT"-test that uses filter factories and ensures they work correctly.

If it is possible to verify this, we can do it in a follow-up issue/PR. It doesn't need to be addressed in this PR.

Reply:
I'll create follow-up issues for areas to test with native AOT enabled. Any code that tests `IsDynamicCodeSupported` is a good candidate.

Reply:
The plan is to undo this anyways. This should be suppression free after we execute https://github.com/dotnet/aspnetcore/issues/45524. There will be new API changes that remove the use of RequestDelegateFactory.